lib/geminabox/remote_cache.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+require "fileutils"
+
+module Geminabox
+  class RemoteCache
+    attr_reader :cache_path, :temp_dir, :gems_dir, :info_dir
+
+    def initialize
+      root = Pathname.new(File.expand_path(Geminabox.data))
+      @cache_path = root / 'remote_cache'
+      @gems_dir = @cache_path / 'gems'
+      @info_dir = @cache_path / 'info'
+      @temp_dir = root / '_temp'
+      ensure_dirs_exist!
+    end
+
+    def flush(file_path)
+      FileUtils.rm_f(cache_path + file_path)
+    end
+
+    def flush_all
+      FileUtils.rm_rf(cache_path)
+      ensure_dirs_exist!
+    end
+
+    def cache(file_path)
+      path = cache_path + file_path
+      write(path, yield) unless path.exist?
+      path
+    end
+
+    def md5(file_path)
+      data = read(file_path)
+      return nil unless data
+
+      Digest::MD5.hexdigest(data)
+    end
+
+    def read(file_path)
+      File.read(cache_path + file_path)
```

Comment:
cache entries are saved in binary mode, it should be safer to read them the same way
```suggestion
      File.binread(cache_path + file_path)
```

Reply:
thx!